yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -83,7 +83,19 @@ private[yarn] class YarnAllocator(
   @volatile private var numExecutorsRunning = 0
   // Used to generate a unique ID per executor
   private var executorIdCounter = 0
-  @volatile private var numExecutorsFailed = 0
+
+  // Queue to store the timestamp of failed executors
+  private val failedExecutorsTimeStamps = new Queue[Long]()
+
+  private var clock: Clock = new SystemClock
+
+  private val executorFailuresValidityInterval = {
+    if (sparkConf.contains("spark.yarn.executor.failuresValidityInterval")) {
```

Comment:
just use sparkConf. getTimeAsMs with the default value of -1L. 
/**
   * Get a time parameter as milliseconds, falling back to a default if not set. If no
   * suffix is provided then milliseconds are assumed.
   */
  def getTimeAsMs(key: String, defaultValue: String):